AI/Nullkiller/Behaviors/BuildingBehavior.cpp

```diff
@@ -49,26 +49,49 @@ Goals::TGoalVec BuildingBehavior::decompose(const Nullkiller * ai) const
 	auto & developmentInfos = ai->buildAnalyzer->getDevelopmentInfo();
 	auto isGoldPressureLow = !ai->buildAnalyzer->isGoldPressureHigh();
 
+	ai->dangerHitMap->updateHitMap();
```

Comment:
May be quite expensive. Treats should not change significantly during our turn. Only in case some enemy hero was killed.

Reply:
void DangerHitMapAnalyzer::updateHitMap()
{
	if(hitMapUpToDate)
		return;

has this at right the beginning. So it won't really do anything if it ran already. it only gets updated here because it wasn't updated before. This then also means it doesn't need to be updated again later.